### DISCUSSION TITLE
Member alignment issue when importing SAF

### DISCUSSION THREAD

**Discussion by tbyrk** [idea-statica/ideastatica-public#467] [Rhino/Grasshopper] [answered]
Hi

Member alignment is wrong when members have differing eccentricities at ends. Member should be sloped but in exported connections it remains horizontal. This will eventually cause unbalanced forces as well.

![SOFiSTiK model](https://github.com/idea-statica/ideastatica-public/assets/162557189/8a954e7d-769d-4cfd-9040-e4d9263059ca)
<img width="493" alt="SAF vector" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/f1cdd428-745e-4805-838a-0f36fd86c785">
<img width="503" alt="SAF eccentricities" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/707bf29a-909f-4869-bfd5-854e2337cb76">
<img width="719" alt="Grasshopper script" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/dfc4fa83-c547-40fc-a498-7c9fdcd295b8">
<img width="1274" alt="Horizontal members" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/e89714a3-3b1c-407e-8f79-8c608b404496">
[Frame.xlsx](https://github.com/idea-statica/ideastatica-public/files/14521218/Frame.xlsx)

I'm trying to feed the SAF to IDEA StatiCa Grasshopper components to eventually mass calculate welded hollow section truss connections where control over eccentricities (gap or overlap) is of critical importance. Members have to be connected to one node otherwise this task becomes impossible with IDEA StatiCa. After calculations I want to deconstruct the connections (maybe some with additional strengthening plates) in Grasshopper and reconstruct them for Tekla Structures.

Workflow - Grasshopper > SOFiSTiK > SAF > Grasshopper > IDEA StatiCa > Grasshopper > Tekla Structures

<img width="943" alt="Goal" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/2b8acc19-c19b-4b7c-80ec-eb7f63c83477">
<img width="578" alt="1993-1-8" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/b7fddb6f-b3f5-43d6-a167-c5a76fe23ecd">

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi @tbyrk, The import into checkbot should be fixed in the next version, v24.0.

**Reply by nathanlukeIDEA** [reply]
However, It will not be shown in Rhino/Grasshopper yet.
